### DISCUSSION TITLE
Invalid default for TIMESTAMP

### DISCUSSION THREAD

**Discussion by garlandcrow** [planetscale/discussion#428] [Bug Reports] [answered]
I am going through a very basic starter tutorial on using planetscale and can't get a simple "created_at" TIMESTAMP column to work. Here is the SQL that I am executing (I did this via the planetscale DB console so that I can remove other possible reasons for the error):

```mysql
CREATE TABLE `posts` (
	`id` varchar(191) PRIMARY KEY NOT NULL,
	`created_at` timestamp(2) NOT NULL DEFAULT (now()),
	`content` varchar(256) NOT NULL
);
```

I have tried `(now())` and `now()` and `CURRENT_TIMESTAMP` and `CURRENT_TIMESTAMP()` and no matter what I get the same error returned:

`target: chirp.-.primary: vttablet: rpc error: code = InvalidArgument desc = Invalid default value for 'created_at' (errno 1067) (sqlstate 42000)`

Is there something different about PS vs msql that means I should use something else to make a default for TIMESTAMP?

**Comment by jgreet** [comment, marked as answer]
Can you try matching the fraction seconds precision for `timestamp` and `now()`? 

This seems to be working for me, but I see what you mean about the difference in behavior between MySQL 8 and PlanetScale. Definitely worth looking in to some more.

```sql
CREATE TABLE `posts` (
	`id` varchar(191) PRIMARY KEY NOT NULL,
	`created_at` timestamp(2) NOT NULL DEFAULT now(2),
	`content` varchar(256) NOT NULL
);
```

**Reply by garlandcrow** [reply]
Ah yea that works when they're the same. I'll pass this info back to the query builder I'm using so they can generate this correctly for PS, thanks!
